plugins/content_distribution/lib/kContentDistributionFlowManager.php

```diff
@@ -30,9 +34,17 @@ public function objectChanged(BaseObject $object, array $modifiedColumns)
 		if($object instanceof entry && $object->getStatus() != entryStatus::DELETED)
 		{
 			if(in_array(entryPeer::STATUS, $modifiedColumns) && $object->getStatus() == entryStatus::READY)
+			{
 				return self::onEntryReady($object);
+			}
+			elseif(in_array(entryPeer::MODERATION_STATUS, $modifiedColumns))
```

Comment:
You should move the new moderation status check here (if(!in_array($entry->getModerationStatus(), self::$validModerationStatuses))).